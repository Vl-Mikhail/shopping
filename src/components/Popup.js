import React from "react";
import PropTypes from "prop-types";
import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
import ItemList from "./ItemList";

class Popup extends React.Component {
  state = {
    modal: false
  };

  toggle = () => {
    this.setState({
      modal: !this.state.modal
    });
  };

  render() {
    const { items, buttonLabel } = this.props;

    return (
      <div>
        <Button color="link" onClick={this.toggle}>
          {buttonLabel}
        </Button>
        <Modal isOpen={this.state.modal} toggle={this.toggle}>
          <ModalHeader toggle={this.toggle}>Список товаров</ModalHeader>
          <ModalBody>
            <ItemList items={items} />
          </ModalBody>
          <ModalFooter>
            <Button color="primary" onClick={this.toggle}>
              Оформить
            </Button>{" "}
            <Button color="secondary" onClick={this.toggle}>
              Отмена
            </Button>
          </ModalFooter>
        </Modal>
      </div>
    );
  }
}

Popup.propTypes = {
  items: PropTypes.array,
  buttonLabel: PropTypes.string
};

Popup.defaultProps = {
  items: [],
  buttonLabel: ""
};

export default Popup;
