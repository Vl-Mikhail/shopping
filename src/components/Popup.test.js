import React from 'react';
import { shallow } from 'enzyme';
import Popup from './Popup';


it('renders without crashing', () => {
  expect(shallow(<Popup/>)).toMatchSnapshot();
});