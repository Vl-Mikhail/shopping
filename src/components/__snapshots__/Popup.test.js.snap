// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Popup
    buttonLabel=""
    items={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          color="link"
          onClick={[Function]}
          tag="button"
/>,
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
                    Object {
                              "mountOnEnter": true,
                              "timeout": 150,
                            }
          }
          centered={false}
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
                    Object {
                              "timeout": 300,
                            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          role="dialog"
          toggle={[Function]}
          zIndex={1050}
>
          <ModalHeader
                    charCode={215}
                    closeAriaLabel="Close"
                    tag="h5"
                    toggle={[Function]}
                    wrapTag="div"
          >
                    Список товаров
          </ModalHeader>
          <ModalBody
                    tag="div"
          >
                    <Unknown
                              items={Array []}
                    />
          </ModalBody>
          <ModalFooter
                    tag="div"
          >
                    <Button
                              color="primary"
                              onClick={[Function]}
                              tag="button"
                    >
                              Оформить
                    </Button>
                     
                    <Button
                              color="secondary"
                              onClick={[Function]}
                              tag="button"
                    >
                              Отмена
                    </Button>
          </ModalFooter>
</Modal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "color": "link",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <ModalHeader
              charCode={215}
              closeAriaLabel="Close"
              tag="h5"
              toggle={[Function]}
              wrapTag="div"
>
              Список товаров
</ModalHeader>,
            <ModalBody
              tag="div"
>
              <Unknown
                            items={Array []}
              />
</ModalBody>,
            <ModalFooter
              tag="div"
>
              <Button
                            color="primary"
                            onClick={[Function]}
                            tag="button"
              >
                            Оформить
              </Button>
               
              <Button
                            color="secondary"
                            onClick={[Function]}
                            tag="button"
              >
                            Отмена
              </Button>
</ModalFooter>,
          ],
          "fade": true,
          "isOpen": false,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "role": "dialog",
          "toggle": [Function],
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "charCode": 215,
              "children": "Список товаров",
              "closeAriaLabel": "Close",
              "tag": "h5",
              "toggle": [Function],
              "wrapTag": "div",
            },
            "ref": null,
            "rendered": "Список товаров",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Unknown
                items={Array []}
/>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "items": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Button
                  color="primary"
                  onClick={[Function]}
                  tag="button"
>
                  Оформить
</Button>,
                " ",
                <Button
                  color="secondary"
                  onClick={[Function]}
                  tag="button"
>
                  Отмена
</Button>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Оформить",
                  "color": "primary",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Оформить",
                "type": [Function],
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Отмена",
                  "color": "secondary",
                  "onClick": [Function],
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Отмена",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            color="link"
            onClick={[Function]}
            tag="button"
/>,
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropTransition={
                        Object {
                                    "mountOnEnter": true,
                                    "timeout": 150,
                                  }
            }
            centered={false}
            fade={true}
            isOpen={false}
            keyboard={true}
            modalTransition={
                        Object {
                                    "timeout": 300,
                                  }
            }
            onClosed={[Function]}
            onOpened={[Function]}
            role="dialog"
            toggle={[Function]}
            zIndex={1050}
>
            <ModalHeader
                        charCode={215}
                        closeAriaLabel="Close"
                        tag="h5"
                        toggle={[Function]}
                        wrapTag="div"
            >
                        Список товаров
            </ModalHeader>
            <ModalBody
                        tag="div"
            >
                        <Unknown
                                    items={Array []}
                        />
            </ModalBody>
            <ModalFooter
                        tag="div"
            >
                        <Button
                                    color="primary"
                                    onClick={[Function]}
                                    tag="button"
                        >
                                    Оформить
                        </Button>
                         
                        <Button
                                    color="secondary"
                                    onClick={[Function]}
                                    tag="button"
                        >
                                    Отмена
                        </Button>
            </ModalFooter>
</Modal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "color": "link",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "backdrop": true,
            "backdropTransition": Object {
              "mountOnEnter": true,
              "timeout": 150,
            },
            "centered": false,
            "children": Array [
              <ModalHeader
                charCode={215}
                closeAriaLabel="Close"
                tag="h5"
                toggle={[Function]}
                wrapTag="div"
>
                Список товаров
</ModalHeader>,
              <ModalBody
                tag="div"
>
                <Unknown
                                items={Array []}
                />
</ModalBody>,
              <ModalFooter
                tag="div"
>
                <Button
                                color="primary"
                                onClick={[Function]}
                                tag="button"
                >
                                Оформить
                </Button>
                 
                <Button
                                color="secondary"
                                onClick={[Function]}
                                tag="button"
                >
                                Отмена
                </Button>
</ModalFooter>,
            ],
            "fade": true,
            "isOpen": false,
            "keyboard": true,
            "modalTransition": Object {
              "timeout": 300,
            },
            "onClosed": [Function],
            "onOpened": [Function],
            "role": "dialog",
            "toggle": [Function],
            "zIndex": 1050,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "charCode": 215,
                "children": "Список товаров",
                "closeAriaLabel": "Close",
                "tag": "h5",
                "toggle": [Function],
                "wrapTag": "div",
              },
              "ref": null,
              "rendered": "Список товаров",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Unknown
                  items={Array []}
/>,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "items": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Button
                    color="primary"
                    onClick={[Function]}
                    tag="button"
>
                    Оформить
</Button>,
                  " ",
                  <Button
                    color="secondary"
                    onClick={[Function]}
                    tag="button"
>
                    Отмена
</Button>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Оформить",
                    "color": "primary",
                    "onClick": [Function],
                    "tag": "button",
                  },
                  "ref": null,
                  "rendered": "Оформить",
                  "type": [Function],
                },
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Отмена",
                    "color": "secondary",
                    "onClick": [Function],
                    "tag": "button",
                  },
                  "ref": null,
                  "rendered": "Отмена",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
